plugins/modules/site_intent.py

```diff
@@ -774,10 +783,47 @@ def get_diff_deleted(self, config):
         site_exists = self.have.get("site_exists")
 
         if site_exists:
```

Comment:
We can check if site not exists and return from here. So we don't need to have much code in site_exists.

```
if not site_exists:
    msg = f"Cannot delete Site - {site_name} as it's not found in Cisco DNA Center"
   self.result.update({'changed': False, 
                       'response': msg, 
                       'msg': msg})
   self.log(msg)
   self.status = "success"
   return self

# Check here if the site have the childs then fetch it using get membership API and then sort it
..
..

```

Reply:
added.